jmix-search/search/src/main/java/io/jmix/search/index/impl/IndexConfigurationsChecker.java

```diff
@@ -0,0 +1,102 @@
+import org.elasticsearch.client.indices.GetIndexResponse;
+import org.elasticsearch.cluster.metadata.MappingMetadata;
+import org.elasticsearch.common.settings.Settings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+
+@Component
+public class IndexConfigurationsChecker {
+    private final SearchMappingComparator mappingComparator;
+    private final SearchSettingsComparator settingsComparator;
+
+    public IndexConfigurationsChecker(SearchMappingComparator searchMappingChecker, SearchSettingsComparator settingsComparator) {
+        this.mappingComparator = searchMappingChecker;
+        this.settingsComparator = settingsComparator;
+    }
+
+    private static final Logger log = LoggerFactory.getLogger(IndexConfigurationsChecker.class);
+
+
+    protected ObjectMapper objectMapper = new ObjectMapper();
+
+    public ConfiguarionComparingResult compareConfigurations(IndexConfiguration indexConfiguration, GetIndexResponse indexResponse) {
+        ComparingState mappingState = isIndexMappingActual(indexConfiguration, indexResponse);
+        ComparingState settingsState = isIndexSettingsActual(indexConfiguration, indexResponse);
+
+        return new ConfiguarionComparingResult(
+                mappingState.isReindexingNotRequired() && settingsState.isReindexingNotRequired(),
+                mappingState.isConfigurationUpdateRequired(),
+                settingsState.isConfigurationUpdateRequired()
+        );
+    }
+
+    protected ComparingState isIndexMappingActual(IndexConfiguration indexConfiguration, GetIndexResponse indexResponse) {
+        Map<String, MappingMetadata> mappings = indexResponse.getMappings();
+        MappingMetadata indexMappingMetadata = mappings.get(indexConfiguration.getIndexName());
+        Map<String, Object> searchIndexMapping = indexMappingMetadata.getSourceAsMap();
+        Map<String, Object> applicationMapping = objectMapper.convertValue(
+                indexConfiguration.getMapping(),
+                new TypeReference<Map<String, Object>>() {
+                }
+        );
+        log.debug("Mappings of index '{}':\nCurrent: {}\nActual: {}",
+                indexConfiguration.getIndexName(), applicationMapping, searchIndexMapping);
+        return mappingComparator.compare(searchIndexMapping, applicationMapping);
+
+    }
+
+    protected ComparingState isIndexSettingsActual(IndexConfiguration indexConfiguration, GetIndexResponse indexResponse) {
+        Map<String, Settings> settings = indexResponse.getSettings();
+        Settings currentSettings = settings.get(indexConfiguration.getIndexName());
+        Settings actualSettings = indexConfiguration.getSettings();
+        return settingsComparator.compare(currentSettings, actualSettings);
+    }
+
+    public class ConfiguarionComparingResult {
+        private final boolean compartible;
+        private final boolean mappingMustBeActualized;
```

Comment:
Confusing: 'mappingMustBeActualized' - is it required to recreate index or support update (so it's not 'must', but 'can')? If this point also required to take into account the 'compatible' value - it's hard-to-understand and allow to create result with impossible values combination.

Need to simplify

Reply:
Fixed